### PR TITLE
Line clamp sidebar title to 2 lines

### PR DESCRIPTION
**Pull request by skjnldsv**
Fix https://github.com/nextcloud/server/issues/12003

### CODE REVIEW

File: src/components/AppSidebar/AppSidebar.vue [hunk 1]
```diff
@@ -387,20 +426,60 @@ export default {
 		},
 	},
 
+	updated() {
+		this.$nextTick(this.forceUpdateClamps)
```
Comment (skjnldsv) [inline src/components/AppSidebar/AppSidebar.vue:430]: Temporary hack, for some reason the docs doesn't trigger the open/close properly.
On real life example it seems to works fine :thinking: 

Capture d’écran_2021-03-25_16-39-46

File: src/components/AppSidebar/AppSidebar.vue [hunk 2]
```diff
@@ -387,20 +426,60 @@ export default {
...
+		 *
+		 * @param {HTMLElement} element the sidebar aside element
+		 * @param {Function} done the done appearing callback
+		 */
+		onAppear(element, done) {
+			/**
+			 * The sidebar is opening and the transition is in progress
+			 * @type {HTMLElement}
+			 */
+			this.$emit('open', element)
```
Comment (skjnldsv) [inline src/components/AppSidebar/AppSidebar.vue:457]: ChristophWurst will love this, two proper separate events to manage before and after open :)

**Comment by raimund-schluessler** [comment]
Can we have https://github.com/nextcloud/nextcloud-vue/pull/1767 before, please? 🙈

**Comment by skjnldsv** [comment]
> Can we have #1767 before, please?


Yes, this is still a draft! :)
I actually thought that was in already lol

**Pull request closed without merging**